fdbserver/ApplyMetadataMutation.cpp

```diff
@@ -1037,4 +1046,36 @@ void applyMetadataMutations(SpanID const& spanContext,
                             const VectorRef<MutationRef>& mutations,
                             IKeyValueStore* txnStateStore) {
 	ApplyMetadataMutationsImpl(spanContext, dbgid, arena, mutations, txnStateStore).apply();
-}
\ No newline at end of file
+}
+
+bool containsMetadataMutation(const VectorRef<MutationRef>& mutations) {
+	for (auto const& m : mutations) {
+
+		if (m.type == MutationRef::SetValue && isSystemKey(m.param1)) {
+			if (m.param1.startsWith(globalKeysPrefix) || (m.param1.startsWith(cacheKeysPrefix)) ||
+			    (m.param1.startsWith(configKeysPrefix)) || (m.param1.startsWith(serverListPrefix)) ||
+			    (m.param1.startsWith(storageCachePrefix)) || (m.param1.startsWith(serverTagPrefix)) ||
+			    (m.param1.startsWith(tssMappingKeys.begin)) || (m.param1.startsWith(tssQuarantineKeys.begin)) ||
+			    (m.param1.startsWith(applyMutationsEndRange.begin)) ||
+			    (m.param1.startsWith(applyMutationsKeyVersionMapRange.begin)) ||
+			    (m.param1.startsWith(logRangesRange.begin)) || (m.param1.startsWith(serverKeysPrefix)) ||
+			    (m.param1.startsWith(keyServersPrefix)) || (m.param1.startsWith(cacheKeysPrefix))) {
```

Comment:
Can we make a (/an inlined) wrapper method (maybe in "MutationRef"?) for these checks please?

Reply:
I can expose this in `ApplyMetadataMutation.h`.

Reply:
Ok.